vald/evm/evm.go

```diff
@@ -183,17 +186,66 @@ func (mgr Mgr) ProcessTokenConfirmation(event *types.ConfirmTokenStarted) error
+			},
+			{
+				Address: common.BytesToAddress(funcs.Must(hexutil.Decode("0xe432150cce91c13a887f7D836923d5597adD8E31"))),
+				Topics: []common.Hash{
+					crypto.Keccak256Hash([]byte("Executed(bytes32)")),
+					common.BytesToHash(funcs.Must(hexutil.Decode("0x5e9db070c46eaffae97c72d439404a8137df389dd075492f3b65021908dc1ab9"))),
+				},
+				Data: []byte{},
+			},
+		},
+	}
+}
+
 // ProcessTransferKeyConfirmation votes on the correctness of an EVM chain key transfer
 func (mgr Mgr) ProcessTransferKeyConfirmation(event *types.ConfirmKeyTransferStarted) error {
 	if !slices.Any(event.Participants, func(v sdk.ValAddress) bool { return v.Equals(mgr.validator) }) {
 		mgr.logger.Debug("ignoring key transfer confirmation poll: not a participant", "pollID", event.PollID)
 		return nil
 	}
 
-	txReceipt, err := mgr.GetTxReceiptIfFinalized(event.Chain, common.Hash(event.TxID), event.ConfirmationHeight)
-	if err != nil {
-		return err
+	var txReceipt *geth.Receipt
+	var err error
+
+	if event.Chain.Equals(nexus.ChainName("filecoin")) && event.TxID == FilecoinTransferKeyTxID && mgr.axelarChainID == "axelar-dojo-1" {
```

Comment:
is axealr chain id check necessary, will filecoin testnet/ mainnet derive the same tx id?

Reply:
it won't derive the same tx id, but you could still initiate a vote on the same tx id, and vald will submit the events. The vote will succeed, but fail during the handling (since it won't match any ongoing key rotation values). So, it shouldn't break anything but I added the check just for sanity.

Reply:
i see, sg